src/components/bmfont-text.js

```diff
@@ -0,0 +1,333 @@
+  },
+
+  init: function () {
+    this.texture = new THREE.Texture();
+    this.texture.anisotropy = MAX_ANISOTROPY;
+
+    this.geometry = createTextGeometry();
+
+    this.updateMaterial();
+    this.mesh = new THREE.Mesh(this.geometry, this.material);
+    this.el.setObject3D('bmfont-text', this.mesh);
+  },
+
+  update: function (oldData) {
+    var data = this.coerceData(this.data);
+
+    // decide whether to update font, or just text data
+    if (!oldData || oldData.font !== data.font) {
+      // new font, will also subsequently change data & layout
+      this.updateFont();
+    } else if (this.currentFont) {
+      // new data like change of text string
+      var font = this.currentFont;
+      var textRenderWidth = data.wrappixels || (data.wrapcount * 0.6035 * font.info.size);
+      var options = assign({}, data, { font: font, width: textRenderWidth, lineHeight: data.lineHeight || font.common.lineHeight });
+      this.geometry.update(options);
+      this.updateLayout(data);
+    }
+    // if shader changed, update
+    this.updateMaterial(oldData && {shader: oldData.shader, customShader: oldData.customShader});
+  },
+
+  remove: function () {
+    this.geometry.dispose();
+    this.geometry = null;
+    this.el.removeObject3D('bmfont-text');
+  },
+
+  coerceData: function (data) {
+    // We have to coerce some data to numbers/booleans,
+    // as they will be passed directly into text creation and update
+    data = assign({}, data);
+    if (data.lineHeight !== undefined) {
+      data.lineHeight = parseFloat(data.lineHeight);
+      if (!isFinite(data.lineHeight)) { data.lineHeight = undefined; }
+    }
+    if (data.width !== undefined) {
+      data.width = parseFloat(data.width);
+      if (!isFinite(data.width)) { data.width = undefined; }
+    }
+    return data;
+  },
+
+  updateMaterial: function (oldShader) {
+    var data;
+    var changedShader = (oldShader && oldShader.shader) !== this.data.shader || (oldShader && oldShader.customShader) !== this.data.customShader;
+
+    if (changedShader || this.data.customShader) {
+      data = {
+        side: threeSideFromString(this.data.side),
+        transparent: this.data.transparent,
+        alphaTest: this.data.alphaTest,
+        color: this.data.color,
+        opacity: this.data.opacity,
+        map: this.texture
+      };
+    }
+    if (changedShader) {
+      var shader;
+      if (this.data.shader === 'custom') {
+        var ShaderType = shaders[this.data.customShader].Shader;
+        var shaderObject = this.shaderObject = new ShaderType();
+        shaderObject.el = this.el;
+        shaderObject.init(data);
+        shaderObject.update(data);
+        shader = shaderObject.material;
+        shader.transparent = data.transparent; // apparently this is not set on either init or update
+        this.material = shader;
+        if (this.mesh) { this.mesh.material = this.material; }
+        return;
+      }
+      if (this.data.shader === 'SDF') {
```

Comment:
modified-sdf is customShader.  I couldn't figure out how to do the validation oneOf with a dynamic shader list and pre-existing values at the same time, or else I would have done as shader